apps/web/src/pages/Profile/AccountSettings/NotificationSettings.tsx

```diff
@@ -0,0 +1,262 @@
+import * as React from "react";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import { useIntl } from "react-intl";
+import { useMutation } from "urql";
+
+import {
+  Checkbox,
+  CheckboxOption,
+  Field,
+  Submit,
+} from "@gc-digital-talent/forms";
+import { NotificationFamily, graphql } from "@gc-digital-talent/graphql";
+import { Card } from "@gc-digital-talent/ui";
+import { commonMessages, formMessages } from "@gc-digital-talent/i18n";
+import { toast } from "@gc-digital-talent/toast";
+
+import { dataValuesToFormValues, formValuesToData } from "./utils";
+import { FormValues } from "./types";
+
+export type UpdateNotificationInput = {
+  ignoredEmailNotifications?: [NotificationFamily];
+  ignoredInAppNotifications?: [NotificationFamily];
+};
+
+interface NotificationChecklistProps {
+  id: string;
+  name: string;
+  options: CheckboxOption[];
+  legend: string;
+  subtitle: string;
+}
+
+const NotificationChecklist = ({
+  id,
+  name,
+  options,
+  legend,
+  subtitle,
+}: NotificationChecklistProps) => {
+  return (
+    <Field.Wrapper data-h2-margin-bottom="base(x1)">
+      <Field.Fieldset id="systemMessages">
+        <Field.Legend
+          required={false}
+          data-h2-visually-hidden="base(invisible)"
+        >
+          {legend}
+        </Field.Legend>
+        <div
+          data-h2-display="base(grid)"
+          data-h2-grid-template-columns="base(4fr 1fr)"
+          data-h2-align-items="base(center)"
+        >
+          <div>
+            <p
+              data-h2-font-weight="base(bold)"
+              data-h2-padding-bottom="base(x.5)"
+            >
+              {legend}
+            </p>
+            <p data-h2-font-size="base(caption)">{subtitle}</p>
+          </div>
+          <div
+            data-h2-display="base(flex)"
+            data-h2-justify-content="base(space-between)"
+          >
+            {options.map(({ value, label, disabled }) => {
```

Comment:
I think you'll want to add to the type if you want to do this. https://github.com/GCTC-NTGC/gc-digital-talent/blob/eb2c69a2097d89887f7154fe046910236b503838/packages/forms/src/components/Checklist/Checklist.tsx#L13-L16

```ts
export type CheckboxOption = {
  value: string | number;
  label: string | React.ReactNode;
  disabled?: boolean;
};
```

@esizer any a11y reason not to disable a checkbox?

Reply:
If we don't have to, I'd say avoid it because it always ends up causing more confusion than it saves us. But if we do have to, we should clearly state:

- Why it is disabled
- If it can be enabled by a user action and if so;
    - What they need to do to have be enabled
    
 Honestly, if they can't interact with something and nothing they do can fix that, we should just hide it :woman_shrugging:

Reply:
Okay I fixed the disabled error. However as to the design, that's a question for @substrae.

Reply:
These "disabled" checkboxes shouldn't be semantic checkboxes - they're visual placeholders that represent a setting that can't be changed.